p2p/src/authenticated/discovery/actors/peer/actor.rs

```diff
@@ -128,9 +128,14 @@ impl<E: Spawner + Clock + ReasonablyRealtime + Rng + CryptoRng + Metrics, C: Pub
             move |context| async move {
                 // Set the initial deadline to now to start gossiping immediately
                 let mut deadline = context.current();
+                let mut shutdown = context.stopped();
```

Comment:
As discussed in #2335, we want each actor to explicitly manage its own shutdown rather than just relying on closed channels.